fastify.js

```diff
@@ -466,18 +466,40 @@ function build (options) {
     return it
   }
 
+  function _inject (server, opts, cb) {
+    if (cb) {
+      shot.inject(server, opts, cb)
+      return
+    }
+    return new Promise(resolve => {
+      shot.inject(server, opts, resolve)
+    })
+  }
+
   function inject (opts, cb) {
     if (!shot) throw new Error('"shot" library is not installed: "npm install shot@3 --save-dev"')
 
     if (started) {
-      shot.inject(this, opts, cb)
+      return _inject(this, opts, cb)
+    }
+
+    if (cb) {
+      this.ready(err => {
+        if (err) throw err
+        return _inject(this, opts, cb)
```

Comment:
I think you can avoid defining `_inject()`  and just use `shot.inject` here.

Reply:
Yes, but with `_inject` there's encapsulated the "`shot` logic".

Anyway if you like more the substitution, I'll do it!

Reply:
Yes, it would be less code overall.